assessments/CLI_Circle/earthmover.yml

```diff
@@ -0,0 +1,589 @@
+      - operation: rename_columns
+        columns:
+          {{ subject_name }}: subject_score
+          {{ subject_name }}_date: subject_date
+      
+      # Some subjects have "sets" and a student might be assessed based on one of these sets.
+      # For example, Rapid Letter Naming has 3 sets, each set contains multiple combinations of letters, a student can be assessed on one of these sets.
+      # For these subjects we have to create a "subject_set" column which will tell us what set a student was assessed with.
+      {% if subject_values.sets is not none and subject_values.sets | length %}
+        {% for set in subject_values.sets %}
+          {% for set_category, set_values in set.items() %}
+            {% for set_key, set_value in set_values.items() %}
+          {{ set_key }}_date: subject_set_date
+          {{ set_key }}: subject_set_score
+          cp_{{ set_key }}: subject_set_performance
+
+      - operation: add_columns
+        columns:
+          subject_set_name: {{ set_key }}
+          subject_set_descriptor: {{ set_value }}
+          subject_set: {{ set_category.split('_')[-1] }}
+            {% endfor %}
+          {% endfor %}
+        {% endfor %}
+      {% else %}
+      - operation: add_columns
+        columns:
+          subject_set_name: ''
+          subject_set_descriptor: ''
+          subject_set: ''
+          subject_set_date: ''
+          subject_set_score: ''
+          subject_set_performance: ''
+      {% endif %}
+
+      # Here we deal with subjects that do not have a Cut Point column (Performance indicator).
+      # If the subject_name is not found in the "non_cp_columns" list it will create a subject_performance column with a proper value.
+      # If it is, it will create a null column with the name "subject_performance" so that our keep columns operation does not break.
+      {% if subject_name not in non_cp_columns %}
+      - operation: rename_columns
+        columns:
+          cp_{{ subject_name }}: subject_performance
+      {% else %}
+      - operation: add_columns
+        columns:
+          subject_performance: ''
+      {% endif %}
+
+      # Add the final subject columns: name and descriptor (e.g. "math_v1" and "Math")
+      - operation: add_columns
+        columns:
+          subject_name: {{ subject_name }}
+          subject_descriptor: {{ subject_values.name }}
+
+      # Now we create objective assessment data based on our dictionaries above.
+      # If a subject has objective assessments listed, then it conforms all those objective assessments 
+      # by giving them generic names that can then stack on top of each other e.g. "oa_category_1"..."oa_category_7"
+      {% if subject_values.objective_assessments is not none and subject_values.objective_assessments | length %}
+      {% set cat_numbers=[] -%}
+      {% for objective_assessment in subject_values.objective_assessments %}
+        {% for oa_category, oa_values in objective_assessment.items() %}
```

Comment:
We can use loop index but given that oa_category is built in the dictionary as a key, it may make more sense to use this hard coded value instead:
```
     'pa_v2': {
      'colname': 'pa_v2',
      'descriptor': 'Phonological Awareness',
      'objective_assessments': [
        {'oa_category_1':{'colname':'pa_v2_syllabication_v2', 'descriptor':'Phonological Awareness Syllabication'}},
        {'oa_category_2':{'colname':'pa_v2_alliteration_v2', 'descriptor':'Phonological Awareness Alliteration'}},
        {'oa_category_3':{'colname':'pa_v2_rhyming_i_v2', 'descriptor':'Phonological Awareness Rhyming I'}},
        {'oa_category_4':{'colname':'pa_v2_onset_rime_v2', 'descriptor':'Phonological Awareness Onset-Rime'}}
      ],
      'subject_assess_lang': 'English',
      'subject_has_cp_col': True,
      'oa_has_cp_col': True
    },
```

Reply:
yes but my thought was that by using `loop.index` you remove the need to hard-code, so the dictionary can be simpler:
```
'pa_v2': {
      'colname': 'pa_v2',
      'descriptor': 'Phonological Awareness',
      'objective_assessments': {
        'pa_v2_syllabication_v2':'Phonological Awareness Syllabication',
        'pa_v2_alliteration_v2':'Phonological Awareness Alliteration',
        'pa_v2_rhyming_i_v2':'Phonological Awareness Rhyming I',
        'pa_v2_onset_rime_v2':'Phonological Awareness Onset-Rime'
      },
      'subject_assess_lang': 'English',
      'subject_has_cp_col': True,
      'oa_has_cp_col': True
    },
```

but maybe simpler = less transparent?